### PR TITLE
Use validations for models. Write unit tests for models. Secure app from n+1 problems.

### PR DESCRIPTION
**Pull request by gracekabaghe**
- Use validations for models.
- Write unit tests for models.
- Secure app from n+1 problems.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -1,5 +1,16 @@
 require 'rails_helper'
 
 RSpec.describe Post, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
+  subject { Post.create(author: first_user, title: 'Hello', text: 'This is my first post') }
+
+  before { subject.save }
+
```
Comment (hiromataba) [inline spec/models/post_spec.rb:8]: - [x] Please add more specs for validation as required, it looks something like this :point_down: 

```ruby
describe 'validations' do
    describe 'title' do
      context 'when valid' do
        it { expect(post).to be_valid }
      end
 end
```

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
 RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  subject { User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.') }
+
+  before { subject.save }
+
+  it 'return the three latest posts for a user' do
+    expect(User.recent_post(subject.id).count).to eql 0
+  end
+
```
Comment (hiromataba) [inline spec/models/user_spec.rb]: - [x] Please add more specs for validation as required, it looks something like this :point_down: 

```ruby
describe 'validations' do
  it { should validate_presence_of(:name) }
end
```

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi :wave: :wave: @gracekabaghe  

almost-there

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
### To highlight :medal_sports: 
- Nice PR title and description :heavy_check_mark: 
- Professional README file :+1: 
- No linter errors :heavy_check_mark: 

## Required Changes ♻️

- [ ] Good job adding tests for your models but I think it would be better to add more for validation, you may have a look  at [My repository](https://github.com/hiromataba/Blog-App/tree/add-validation-specs/spec/models) as reference :wink: 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi @gracekabaghe, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Good job working on the requested changes 👍

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**